canary/producer-cpp/src/CanaryConfig.cpp

```diff
@@ -2,115 +2,131 @@
-            nullptr != (thingName = GETENV("AWS_IOT_CORE_THING_NAME"))) {
-            streamName = thingName;
+    if(this->useIotCredentialProvider) {
+        if (nullptr != (this->iotGetCredentialEndpointFile = GETENV(IOT_CORE_CREDENTIAL_ENDPOINT_ENV_VAR)) &&
+            nullptr != (this->certPath = GETENV(IOT_CORE_CERT_ENV_VAR)) &&
+            nullptr != (this->privateKeyPath = GETENV(IOT_CORE_PRIVATE_KEY_ENV_VAR)) &&
+            nullptr != (this->roleAlias = GETENV(IOT_CORE_ROLE_ALIAS_ENV_VAR)) &&
+            nullptr != (this->caCertPath = GETENV(IOT_CORE_CA_CERT_PATH_ENV_VAR)) &&
+            nullptr != (this->thingName = GETENV(IOT_CORE_THING_NAME_ENV_VAR))) {
+            UINT64 fileSize;
+            CHK_STATUS(readFile(this->iotGetCredentialEndpointFile, TRUE, NULL, &fileSize));
+            CHK_ERR(fileSize != 0, STATUS_EMPTY_IOT_CRED_FILE, "Empty credential file");
+            CHK_STATUS(readFile((PCHAR)this->iotGetCredentialEndpointFile, TRUE, this->iotGetCredentialEndpoint, &fileSize));
+            this->iotGetCredentialEndpoint[fileSize - 1] = '\0';
+            this->streamName = this->thingName;
         }
-        else{
-            retStatus = STATUS_NOT_FOUND;
+        else {
             LOG_ERROR("Missing Credential: IOT Credential");
+            CHK(FALSE, STATUS_NOT_FOUND);
         }
-    }
-    else {
-        if(nullptr != (accessKey = GETENV(ACCESS_KEY_ENV_VAR)) &&
-           nullptr != (secretKey = GETENV(SECRET_KEY_ENV_VAR))){
+    } else {
+        if(nullptr != (this->accessKey = GETENV(ACCESS_KEY_ENV_VAR)) &&
+           nullptr != (this->secretKey = GETENV(SECRET_KEY_ENV_VAR))){
             setEnvVarsString(streamName, "CANARY_STREAM_NAME");
         }
         else{
-            retStatus = STATUS_NOT_FOUND;
             LOG_ERROR("Missing Credential: AWS Credential");
+            CHK(FALSE, STATUS_NOT_FOUND);
         }
     }
-    if(STATUS_SUCCEEDED(retStatus)){
-        setEnvVarsString(canaryRunScenario, "CANARY_RUN_SCENARIO");
-        setEnvVarsString(streamType, "CANARY_STREAM_TYPE");
-        setEnvVarsString(canaryLabel, "CANARY_LABEL");
-        setEnvVarsString(cpUrl, "CANARY_CP_URL");
-
-        setEnvVarsInt(&fragmentSize, "CANARY_FRAGMENT_SIZE");
-        setEnvVarsInt(&canaryDuration, "CANARY_DURATION_IN_SECONDS");
-        setEnvVarsInt(&bufferDuration, "CANARY_BUFFER_DURATION");
-        setEnvVarsInt(&storageSizeInBytes, "CANARY_STORAGE_SIZE");
-        setEnvVarsInt(&testVideoFps, "CANARY_FPS");
-
-        defaultRegion = GETENV(DEFAULT_REGION_ENV_VAR);
-        sessionToken = GETENV(SESSION_TOKEN_ENV_VAR);
-
-        LOG_DEBUG("CANARY_STREAM_NAME: " << streamName);
-        LOG_DEBUG("CANARY_RUN_SCENARIO: " << canaryRunScenario);
-        LOG_DEBUG("CANARY_STREAM_TYPE: " << streamType);
-        LOG_DEBUG("CANARY_LABEL: " << canaryLabel);
-        LOG_DEBUG("CANARY_CP_URL: " << cpUrl);
-        LOG_DEBUG("CANARY_FRAGMENT_SIZE: " << fragmentSize);
```

Comment:
We don't need cpUrl?

Reply:
We set this through env in CPP SDK for kvssink